nexus_constructor/main_window.py

```diff
@@ -202,54 +208,16 @@ def _load_current_entry():
         self.entries_dialog.show()
 
     def set_button_state(self):
-        indices = self.component_tree_view.selectedIndexes()
-        if len(indices) != 1:
-            self.delete_action.setEnabled(False)
-            self.duplicate_action.setEnabled(False)
-            self.new_rotation_action.setEnabled(False)
-            self.new_translation_action.setEnabled(False)
-            self.create_link_action.setEnabled(False)
-            self.zoom_action.setEnabled(False)
-        else:
-            selected_object = indices[0].internalPointer()
-            self.zoom_action.setEnabled(isinstance(selected_object, Component))
-            if isinstance(selected_object, Component) or isinstance(
-                selected_object, Transformation
-            ):
-                self.delete_action.setEnabled(True)
-                self.duplicate_action.setEnabled(True)
-                self.edit_component_action.setEnabled(True)
-            else:
-                self.delete_action.setEnabled(False)
-                self.duplicate_action.setEnabled(False)
-                self.edit_component_action.setEnabled(False)
-            if isinstance(selected_object, LinkTransformation):
-                self.new_rotation_action.setEnabled(False)
-                self.new_translation_action.setEnabled(False)
-                self.delete_action.setEnabled(True)
-            else:
-                self.new_rotation_action.setEnabled(True)
-                self.new_translation_action.setEnabled(True)
-
-            if isinstance(selected_object, Component):
-                if not hasattr(selected_object, "stored_transforms"):
-                    selected_object.stored_transforms = selected_object.transforms
-                if not selected_object.stored_transforms.has_link:
-                    self.create_link_action.setEnabled(True)
-                else:
-                    self.create_link_action.setEnabled(False)
-            elif isinstance(selected_object, TransformationsList):
-                if not selected_object.has_link:
-                    self.create_link_action.setEnabled(True)
-                else:
-                    self.create_link_action.setEnabled(False)
-            elif isinstance(selected_object, Transformation):
-                if not selected_object.parent.has_link:
-                    self.create_link_action.setEnabled(True)
-                else:
-                    self.create_link_action.setEnabled(False)
-            else:
-                self.create_link_action.setEnabled(False)
+        set_button_state(
```

Comment:
might be worth renaming this to `_set_button_state` to avoid confusion with the helper method.